src/app/shared/entry.ts

```diff
@@ -175,34 +175,37 @@ export abstract class Entry implements OnInit, OnDestroy, AfterViewInit {
     });
   }
 
-  public selectVersion(versions, urlVersion, defaultVersion, selectedVersion): any {
-    let useFirstTag = true;
-    let urlTagExists = false;
-    // Determine which tag to select
-    for (const item of versions) {
-      // If a tag is specified in the URL then use it
-      if (urlVersion !== null) {
-        if (item.name === urlVersion) {
-          selectedVersion = item;
-          useFirstTag = false;
-          urlTagExists = true;
-          break;
-        }
-      } else if (defaultVersion !== null && !urlTagExists) {
-        // If the tool has a default version then use it
-        if (item.name === defaultVersion) {
-          selectedVersion = item;
-          useFirstTag = false;
-          break;
-        }
+  /**
+   * Given an array of WorkflowVersions or Tag, decide which one to return (displayed to the user)
+   * 1. If the URL specifies a specific Tag or WorkflowVersion, choose to display that one.  Otherwise,
+   * 2. If the default version is specified, choose to display the default version.  Otherwise,
+   * 3. Choose the last version in the array (the last updated version?)
+   * @param {((Array<WorkflowVersion | Tag>))} versions
+   * @param {string} urlVersion
+   * @param {string} defaultVersion
+   * @param {((WorkflowVersion | Tag))} selectedVersion
+   * @returns {((WorkflowVersion | Tag))}
+   * @memberof Entry
+   */
+  public selectVersion(versions: (Array<WorkflowVersion | Tag>), urlVersion: string, defaultVersion: string,
```

Comment:
This is supposed to be (Array<WorkflowVersion> | Array<Tag>), no idea why errors appear when attempted to do so

Reply:
Probably makes more sense as a comment in code for the next developer that looks at this